presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -458,11 +495,44 @@ public QueryInfo createQuery(SessionContext sessionContext, String query)
         queries.put(queryId, queryExecution);
 
         // start the query in the background
-        resourceGroupManager.submit(statement, queryExecution, queryExecutor);
+        resourceGroupManager.submit(statement, queryExecution, group, queryExecutor);
 
         return queryInfo;
     }
 
+    private Optional<String> getQueryType(String query)
+    {
+        Statement statement = sqlParser.createStatement(query, new ParsingOptions(AS_DECIMAL));
+        QueryExecutionFactory<?> queryExecutionFactory = executionFactories.get(statement.getClass());
+
+        if (queryExecutionFactory instanceof DataDefinitionExecutionFactory) {
+            return Optional.of(DATA_DEFINITION.name());
+        }
+
+        if (queryExecutionFactory instanceof SqlQueryExecutionFactory) {
```

Comment:
This code already exists `SqlQueryExecution`. We definitely don't want two copies of this.

Reply:
Can we remove the `getQueryType()` method from `QueryExecution` now?

Reply:
@electrum : Yes we already removed it in https://github.com/prestodb/presto/pull/10264/commits/e70c75313f800da4a4dc44fc36112bfb0ff77140